src/PhpHooks/Utility/GitUtility.php

```diff
@@ -23,18 +29,50 @@ public static function getGitDir()
     /**
      * Thanks to @sascha (https://github.com/sascha-seyfert)
      *
+     * @param string $listType Valid list types are stage, unstaged and untracked
      * @return array
      */
-    public static function extractFiles()
+    public static function extractFiles($listType = 'staged')
     {
         $output = array();
-        exec("git diff --cached --name-status --diff-filter=ACMR", $output);
 
-        $files = array_map(function($v) {
+        switch ($listType) {
```

Comment:
This is an indentifier that the method violate single responsibility.
Why not three methods for three use cases?